documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc

```diff
@@ -0,0 +1,1946 @@
+}
+----
+
+[source, JSON]
+----
+{
+    "version": "1",
+    "title": "On the merits of NoSQL",
+    "description": "This paper discuss why NoSQL will save the world for good"
+}
+----
+====
+
+Redis doesn't have a concept of "tables";
+Instead all values are stored in a unique key.
+Thus Hibernate OGM needs to add two additional attributes:
+
+.Rename field and collection using @Table and @Column
+====
+[source, JAVA]
+----
+@Entity
+@Table(name="Article")
+public class News {
+
+    @Id
+    @Column(name="code")
+    private String id;
+
+    @Version
+    @Column(name="revision")
+    private int revision;
+
+    private String title;
+
+    @Column(name="desc")
+    private String description;
+
+    //getters, setters ...
+}
+----
+
+[source, JSON]
+----
+{
+    "revision": 1,
+    "title": "On the merits of NoSQL",
+    "desc": "This paper discuss why NoSQL will save the world for good"
+}
+----
+====
+
+===== Identifiers
+
+Redis keys are derived from the Entity name and its Id separated by a colon (`:`).
+String-based Id's are used directly within the key, non-string keys are encoded to JSON.
+You can use any persistable Java type as identifier type, e.g. `String` or `long`.
+Hibernate OGM will convert the `@Id` property into a part of the key name
+so you can name the entity id like you want.
+
+====
+[source, JAVA]
+----
+@Entity
+public class News {
+
+    @Id
+    @Column
+    private long id;
+
+    // fields, getters, setters ...
+}
+----
+
+Key-Scheme for `News` entity with an Id of `42`
+
+[source, JSON]
+----
+  News:42
+----
+
+[source, JAVA]
+----
+@Entity
+@Table(name="Article")
+public class News {
+
+    @Id
+    @Column(name="code")
+    private String id;
+
+    // fields, getters, setters ...
+}
+----
+
+Key-Scheme for `News` entity with an Id of `breaking-news`
+
+[source, JSON]
+----
+Article:breaking-news
+----
+====
+
+Note that you also can work with embedded ids (via `@EmbeddedId`),
+Composite Id's are mapped to a JSON object containing keys and values.
+Hibernate OGM thus will create a concatenated representation of the embedded id's properties in this case.
+The columns are sorted in alphabetical order to guarantee the same order.
+
+.Entity with @EmbeddedId
+====
+[source, JAVA]
+----
+@Entity
+public class News {
+
+    @EmbeddedId
+    private NewsID newsId;
+
+    // getters, setters ...
+}
+
+@Embeddable
+public class NewsID implements Serializable {
+
+    private String title;
+    private String author;
+
+    // getters, setters ...
+}
+----
+
+Resulting key:
+[source, JSON]
+----
+News:{"newsId.author": "Guillaume, "newsId.title": "How to use Hibernate OGM ?"},
+----
+====
+
+===== Identifier generation strategies
+
+You can assign id values yourself or let Hibernate OGM generate the value using the
+[classname]`@GeneratedValue` annotation.
+
+Two main strategies are supported:
+
+1. <<redis-table-id-generation-strategy, TABLE>>
+2. <<redis-sequence-id-generation-strategy, SEQUENCE>>
```

Comment:
Would a sequence more "naturally" mapped to a top-level key instead? It also should help avoiding contention when accessing several sequences in parallel.